lims.rst

```diff
@@ -0,0 +1,137 @@
+LIMS API
+========
+
+
+Data structures
+---------------
+
+.. code:: python
+
+    @dataclass
+    class LIMSProposal:
+    """
+    """
+        pass
+        #TDB
+
+    @dataclass
+    class LIMSLaboratory:
+    """
+    """
+        pass
+        #TDB
+
+    @dataclass
+    class LIMSSession:
+    """
+    """
+        pass
+        #TDB
+
+
+
+API Functions
+-------------
+
+
+.. code:: python
+
+    from queue import DataCollectionTask
+
+    def get_proposal_by_username(str:username) -> List[LIMSProposal]:
+        """
+        :param str username: username
+        :returns: The list proposals belonging to username
+        :rtype: List[LIMSProposal]
+        """
+        pass
+
+    def get_proposal(str:code, str:number) -> LIMSProposal:
+        """
+        :param str code: Proposal code
+        :param str number: Proposal number
+        :returns: The proposal belonging to the proposal with code <code>
+                  and number <number>
+        :rtype: LIMSProposal
+        """
+        pass
+   
+    def get_local_contact(str: session_id) -> LIMSPerson:
+        """
+        :param str session_id: session id
+        :returns: The local contact for session with session id <session_id>
+        """
+        pass
+
+    def get_session(str: proposal_id) -> LIMSSession:
+        """
+        :param str proposal_id: proposal id
+        :returns: The current session, if any, for proposal with id <proposal_id>
+        """
+        pass
+
+    def get_sample_list(str: session_id) > List[LimsSample]:
+        """
+        :param str session_id: session id
+        :returns: List of samples for session with id <session_id>
+        :rtype: List[LimsSample]
+        """
+        pass
+
+    def get_sample(str: sample_id) -> LimsSample:
+        """
+        :param str sample_id: 
+        :returns: Sample with id <sample_id>
+        :rtype: LimsSample
+        """
+        pass
+
+    def get_task_url(str: task_id) -> str:
```

Comment:
It will return a link to ispyb, exi or synchweb? I guess the root of the path will come from the xml.

Reply:
Yes, well it will return the link that the LIMS used returns ...

Reply:
The idea is to return a url to the task within the LIMS application used, i.e. EXI.